.travis.yml

```diff
@@ -18,47 +18,102 @@ os:
+  - os: osx
+    env: LATEST_CLANG=true TESTS=fuzz
+  - os: osx
+    env: TESTS=sawDRBG SAW=true
+  - compiler: gcc
+    env: TESTS=sawDRBG SAW=true
+  - os: osx
+    env: TESTS=sawHMACFailure SAW=true
+  - compiler: gcc
+    env: TESTS=sawHMACFailure SAW=true
+  #This exception is because the test isn't finished yet, remove to turn on DRBG Saw tests
+  - compiler: clang
+    env: TESTS=sawDRBG SAW=true
+
+    
+
 before_install:
   # Install GCC 6 if on OSX
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then   brew update                ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then   brew tap homebrew/versions ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then   brew install gcc6          ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then   brew install gnu-indent    ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then   brew install cppcheck      ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then   .travis/install_osx_dependencies.sh ; fi
   # Set GCC 6 as Default on both Ubuntu and OSX
   - alias gcc=$(which gcc-6)
   # Install latest version of clang, clang++, and llvm-symbolizer and add them to beginning of PATH. Needed for fuzzing.
-  - (.travis/install_clang.sh `pwd`/clang-download `pwd`/clang-latest $TRAVIS_OS_NAME) && export PATH=`pwd`/clang-latest/bin:$PATH
-
-  # Install SAW on Linux only 
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then curl http://saw.galois.com/builds/nightly/saw-0.2-2016-07-22-Ubuntu14.04-64.tar.gz > saw.tar.gz; tar -xzf saw.tar.gz; fi
-  - export PATH=$PATH:$PWD/saw-0.2-2016-07-22-Ubuntu14.04-64/bin
+  - if [[ "$LATEST_CLANG" == "true" ]]; then (.travis/install_clang.sh `pwd`/clang-download `pwd`/clang-latest $TRAVIS_OS_NAME) && export PATH=`pwd`/clang-latest/bin:$PATH ; fi
 
-  # Install Z3
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo add-apt-repository ppa:hvr/z3 -y > /dev/null; sudo apt-key -qq update /dev/null; sudo apt-get -qq update; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get -qq install z3 -y --force-yes; fi
+  # Install SAW with Z3 on Linux only 
+  - if [[ "$SAW" == "true" ]]; then .travis/install_saw.sh ; fi
+  - export PATH=$PATH:$PWD/saw-0.2-2016-11-02-Ubuntu14.04-64/bin:$PWD/z3
 
 install:
   # Download and Install LibFuzzer
-  - .travis/install_libFuzzer.sh `pwd`/fuzz_dependencies/libFuzzer-download `pwd`/fuzz_dependencies $TRAVIS_OS_NAME
-  # Download and Install Openssl 1.1.0
+  - if [[ "$TESTS" == "fuzz" ]]; then .travis/install_libFuzzer.sh `pwd`/fuzz_dependencies/libFuzzer-download `pwd`/fuzz_dependencies $TRAVIS_OS_NAME ; fi
+  # Download and Install Openssl
   - .travis/install_openssl_1_1_0.sh `pwd`/libcrypto-build `pwd`/libcrypto-root $TRAVIS_OS_NAME > /dev/null
   # Install python linked with our compiled Openssl for integration tests
-  - sudo "PATH=$PATH" .travis/install_python.sh `pwd`/libcrypto-root > /dev/null
+  - if [[ "$BUILD_S2N" == "true" ]]; then sudo "PATH=$PATH" .travis/install_python.sh `pwd`/libcrypto-root > /dev/null ; fi
```

Comment:
I think you can add `&& "$TESTS" == "integration"` to the conditional here. We're only building the source python to run integration tests